src/character.cpp

```diff
@@ -4948,12 +4950,14 @@ Character::comfort_level Character::base_comfort_value( const tripoint &p ) cons
             comfort -= g->m.move_cost( p );
         }
 
-        auto items = g->m.i_at( p );
-        for( auto &items_it : items ) {
+        const map_stack items = g->m.i_at( p );
+        for( const item &items_it : items ) {
             if( items_it.has_flag( "SLEEP_AID" ) ) {
                 // Note: BED + SLEEP_AID = 9 pts, or 1 pt below very_comfortable
                 comfort += 1 + static_cast<int>( comfort_level::slightly_comfortable );
```

Comment:
I can't test it right now, but I'm pretty sure there's a bug in here. One can put one `pillow` inside a vehicle and yet another under it, thus getting the bonus from both pillows (and two messages as a symptom).